crates/ty_python_semantic/src/semantic_index/definition.rs

```diff
@@ -1033,12 +1041,37 @@ impl ImportFromDefinitionKind {
 #[derive(Clone, Debug, get_size2::GetSize)]
 pub struct ImportFromSubmoduleDefinitionKind {
     node: AstNodeRef<ast::StmtImportFrom>,
+    module: AstNodeRef<ast::Identifier>,
+    module_index: usize,
 }
 
 impl ImportFromSubmoduleDefinitionKind {
     pub fn import<'ast>(&self, module: &'ast ParsedModuleRef) -> &'ast ast::StmtImportFrom {
         self.node.node(module)
     }
+
+    pub fn module<'ast>(&self, module: &'ast ParsedModuleRef) -> &'ast ast::Identifier {
+        self.module.node(module)
+    }
+
+    pub fn target_range(&self, module: &ParsedModuleRef) -> TextRange {
+        let module_ident = self.module(module);
+        let module_str = module_ident.as_str();
+        let Some(component_str) = module_str.split('.').nth(self.module_index) else {
+            // This shouldn't happen but just in case, provide a safe default
+            return module_ident.range();
+        };
+        let base_addr = module_str.as_ptr().addr();
+        let component_addr = component_str.as_ptr().addr();
+        let offset = base_addr.saturating_sub(component_addr);
```

Comment:
Why do we need to subtract pointers here?

Can't we do

```rust
let offset = module_str.text_len() - component_str.text_len()
```

which also removes the need for `TextSize::try_from`